metadata/metadata.go

```
@@ -2795,6 +2880,105 @@ func (serv *MetadataServer) CreateSourceVariant(ctx context.Context, variantRequ
+	if err != nil {
+		logger.Errorw("Failed to lookup provider resource", "error", err)
+		return err
+	}
+
+	providerProto, ok := sourceProvider.Proto().(*pb.Provider)
+	if !ok {
+		logger.Errorw("Provider resource is not a provider proto")
+		return fferr.NewInternalErrorf("provider resource is not a provider proto")
+	}
+
+	fv.OfflineStoreProvider = providerProto.Name
+	logger.Debugw("Backfilled offline store provider for feature", "provider", providerProto.Name)
+
+	return nil
+}
+
+func (serv *MetadataServer) fillOfflineStoreLocations(ctx context.Context, fv *pb.FeatureVariant, logger logging.Logger) error {
+	if len(fv.GetOfflineStoreLocations()) > 0 {
+		logger.Debug("Offline store locations are already set")
+		return nil
+	}
+
+	providerName := fv.GetOfflineStoreProvider()
+	provider, err := serv.lookup.Lookup(ctx, ResourceID{Name: providerName, Type: PROVIDER})
+	if err != nil {
+		logger.Errorw("Failed to lookup provider resource", "error", err)
+		return err
+	}
+
+	providerProto, ok := provider.Proto().(*pb.Provider)
+	if !ok {
+		logger.Errorw("Provider resource is not a provider proto")
+		return fferr.NewInternalErrorf("provider resource is not a provider proto")
+	}
+
+	localizer, err := GetLocalizer(pt.Type(providerProto.GetType()), providerProto.GetSerializedConfig())
+	if err != nil {
+		logger.Warnw("Failed to get localizer, offline store locations will not be updated", "error", err)
+		return nil
+	}
+
+	// Localize the feature variant and create a location entry
+	location, err := localizer.LocalizeFeatureVariant(fv)
+	if err != nil {
+		logger.Errorw("Failed to localize transformation location", "provider", providerName, "error", err)
+		return err
+	}
+
+	locProto := location.Proto()
+	if locProto != nil {
+		fv.OfflineStoreLocations = append(fv.GetOfflineStoreLocations(), location.Proto())
+	} else {
+		logger.Errorw("Location proto is nil, not adding to offline store locations")
```

I'm swallowing the error here and skipping adding anything to locations so that when the localizer is fixed (and it doesn't return nilLocation in the case of spark) it'll just start working